source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -377,9 +416,104 @@ TEST_F(nifgenDriverApiTest, ResetInterchangeCheck_ResetsSuccessfully)
   ::grpc::Status status = GetStub()->ResetInterchangeCheck(&context, request, &response);
 
   EXPECT_TRUE(status.ok());
-  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiFgenDriverApiTest, SendSoftwareEdgeTrigger_TriggersSuccessfully)
+{
+  ::grpc::ClientContext context;
+  fgen::SendSoftwareEdgeTriggerRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_trigger(fgen::Trigger::TRIGGER_NIFGEN_VAL_SCRIPT_TRIGGER);
+  request.set_trigger_id("ScriptTrigger0");
+  fgen::SendSoftwareEdgeTriggerResponse response;
+  ::grpc::Status status = GetStub()->SendSoftwareEdgeTrigger(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
 }
 
+TEST_F(NiFgenDriverApiTest, AllocateNamedWaveform_WriteNamedWaveformF64_WaveformWrittenSuccessfully)
+{
+  std::string channel_name = "0";
+  std::string waveform_name = "TestWaveform";
+  int waveform_size = 5;
+  set_int32_attribute("", fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_OUTPUT_MODE, fgen::OutputMode::OUTPUT_MODE_NIFGEN_VAL_OUTPUT_SCRIPT);
+  ::grpc::ClientContext alloc_wfm_context;
```

Comment:
Similar to `export_attribute_configuration_buffer` and `import_attribute_configuration_buffer` helpers, `AllocateNamedWaveform` and `WriteNamedWaveformF64` should also be moved to helper methods. It would improve readability of test methods significantly when you don't have these additional lines for setting up of request objects inline with rest of the test code.